src/VisualStudio/Core/Def/SolutionBuildMonitor.cs

```diff
@@ -51,7 +62,7 @@ private void SolutionBuildingContextChanged(object sender, UIContextChangedEvent
                         _operation.Dispose();
```

Comment:
operation can be interleaved. you should put operation in a map or something with operation as a key.